src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,232 @@
+				}
+			} else {
+				throw new DukeException("I don't know which to unmark!");
+			}
+			break;
+		}
+	}
+
+	/**
+	 * Adds task into the arraylist.
+	 *
+	 * @param task The task added into the arraylist.
+	 */
+	public void addDetailedTask(Task task) {
+		System.out.println("-----------------------------------------------");
+		System.out.println("Got it. I've added this task:");
+		System.out.println(task);
+		System.out.println("Now you have " + this.list.size() +" tasks in the list.");
+		System.out.println("-----------------------------------------------");
+	}
+
+	/**
+	 * Adds the right type of task to the list.
+	 *
+	 * @param type     Type of task.
+	 * @param parts    The string response broken down into its keywords.
+	 * @throws DukeException If the keywords are missing or invalid.
+	 */
+	public void addTaskType(Parser.Type type, String[] parts) throws DukeException {
+		String part2;
+		switch (type) {
+		case DEADLINE:
+			/** no task declared */
+			if (parts.length <= 1) {
+				throw new DukeException("There's no deadline task!");
+			}
+			part2 = parts[1];
+			String[] deadlineParts =  part2.split(" /by ", 2);
+			if (deadlineParts[0].equals("")){
+				throw new DukeException("There's no deadline task!");
+			}
+			/** no declaration of deadline time */
+			if (deadlineParts.length <= 1) {
+				throw new DukeException("You didn't specify the deadline! Please use /by.");
+			}
+			DeadlineTask deadline = new DeadlineTask(deadlineParts[0], deadlineParts[1]);
+			this.list.add(deadline);
+			addDetailedTask(deadline);
+			break;
+		case TODO:
+			/** no task declared */
+			if (parts.length <= 1) {
+				throw new DukeException("There's no todo task!");
+			}
+			part2 = parts[1];
+			TodoTask todo = new TodoTask(part2);
+			this.list.add(todo);
+			addDetailedTask(todo);
+			break;
+		case EVENT:
+			/** no task declared */
+			if (parts.length <= 1) {
+				throw new DukeException("There's no event task!");
+			}
+			part2 = parts[1];
+			String[] eventParts = part2.split(" /at ", 2);
+			if (eventParts[0].equals("")){
+				throw new DukeException("There's no event task!");
+			}
+			/** no declaration of event time */
+			if (eventParts.length <= 1) {
+				throw new DukeException("You didn't specify the event time! Please use /at.");
+			}
+			EventTask event = new EventTask(eventParts[0], eventParts[1]);
+			this.list.add(event);
+			addDetailedTask(event);
+			break;
+		}
+	}
+
+	/**
+	 * Deletes task from the list.
+	 *
+	 * @param number The index to be removed.
+	 */
+	public void deleteTask(int number) {
+		Task temp = this.list.get(number - 1);
+		this.list.remove(number - 1);
+		System.out.println("-----------------------------------------------");
+		System.out.println("Noted. I've removed this task:");
+		System.out.println(temp);
+		System.out.println("Now you have " + this.list.size() +" tasks in the list.");
+		System.out.println("-----------------------------------------------");
```

Comment:
Perhaps you can consider migrating these print statements into Ui.java as well for a better organisation of classes.